srcs/tokenizer.c

```diff
@@ -0,0 +1,111 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   tokenizer.c                                        :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: jsaariko <jsaariko@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2020/10/15 11:38:56 by jsaariko      #+#    #+#                 */
+/*   Updated: 2020/10/16 14:53:32 by jsaariko      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+#include "parse.h"
+#include "token_table.h"
+
+int		recognize_special_char(char *input, int j)
+{
+	int i;
+
+	i = 0;
+	while (i < token_table_size)
+	{
+		if (ft_strncmp(input + j, g_token_table[i].token, g_token_table[i].len) == 0)
+			return (g_token_table[i].len);
+		i++;
+	}
+	return (0);
+}
+
+void	separate_special_char(char *input, t_token **head, int *j, int *k)
+{
+	int		len;
+	t_token	*token;
+
+	len = recognize_special_char(input, *j);
+	if (len > 0)
+	{
+		token = create_token(input, *k, *j - *k);
+		if (token != NULL)
+			add_token(head, token);
+		token = create_token(input, *j, len);
+		if (token != NULL)
+			add_token(head, token);
```

Comment:
Are you creating two tokens, one for the special character /separator, and one for the token that follows?

Reply:
One for specialchar and one for everything before from last copied part